src/udev/udevadm-hwdb.c

```diff
@@ -646,9 +655,12 @@ static int adm_hwdb(struct udev *udev, int argc, char *argv[]) {
                 }
                 STRV_FOREACH(f, files) {
                         log_debug("reading file '%s'", *f);
-                        import_file(udev, trie, *f);
+                        if (import_file(udev, trie, *f) < 0)
+                            rc = EXIT_FAILURE;
```

Comment:
If `import_file` fails, you jump to `out`, so hwdb.bin is **not** written.

Please use 8 spaces of indentation everywhere (see CODING_STYLE).